ci-operator/step-registry/cucushift/installer/check/ibmcloud/cucushift-installer-check-ibmcloud-commands.sh

```diff
@@ -12,8 +12,49 @@ function checkProviderType {
     providerType=$(yq-go r ${platformStatusFile} 'status.platformStatus.ibmcloud.providerType')
     if [[ ${providerType} != "VPC" ]]; then
         echo "ERROR: ${providerType} is not expected providerType [VPC]!!"
+        check_result=1
         return 1
     else 
+        echo "checkProviderType PASS"
+        return 0
+    fi
+}
+
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    "${IBMCLOUD_CLI}" config --check-version=false
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+#OCP-60816 - [IPI-on-IBMCloud] install cluster under BYON with a different resource group
+function getResources() {
+    "${IBMCLOUD_CLI}" resource service-instances --type all --output JSON | jq -r '.[]|.name+" "+.resource_id' | sort
+}
+
+function checkResourcesInVPC {
+    preResourceFile="${SHARED_DIR}/vpc_resources"
+    provisioned_rg_file="${SHARED_DIR}/ibmcloud_resource_group"
+    resource_group=$(cat "${provisioned_rg_file}")
+    "${IBMCLOUD_CLI}" target -g ${resource_group}
+    curResourceFile=$(mktemp)
+    getResources > ${curResourceFile}
+    echo "cmp ${curResourceFile} ${preResourceFile} ..."
+    cmpRst=$(cmp ${curResourceFile} ${preResourceFile})
+    rst=$?
+    if [ -n "$cmpRst" ]; then
```

Comment:
In this if block, the 1st condition is using `cmpRst` for condition, while in the 2nd `elif`, you are using return value for condition, suggest to use only one element for the condition, command output  or return value.